api/envoy/config/core/v3/protocol.proto

```diff
@@ -331,6 +331,10 @@ message Http1ProtocolOptions {
   // If set, this overrides any HCM :ref:`stream_error_on_invalid_http_messaging
   // <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.stream_error_on_invalid_http_message>`.
   google.protobuf.BoolValue override_stream_error_on_invalid_http_message = 7;
+
+  // Allows sending fully qualified URLs when proxying the first line of the
+  // response. By default, Envoy will only send the path components in the first line.
```

Comment:
Can you elaborate on where the host comes from? Presumably this is the request host (subject to rewriting), not the logical DNS entry etc.